In [1]:
#import libs
import os, glob
import pandas as pd
import seaborn as sns
import plotly.express as px
import matplotlib.pyplot as plt
import numpy as np
from datetime import date, timedelta
import warnings
warnings.simplefilter("ignore")

In [2]:
#Merge CSVs
path = "../data/raw/log"
all_files = glob.glob(os.path.join(path, "*.csv"))
df_from_each_file = (pd.read_csv(f, sep=',') for f in all_files)
df_merged   = pd.concat(df_from_each_file, ignore_index=True)

In [4]:
df = df_merged.copy()
df.head(2)

Severity                          Name Alarm Source Occurred On (NT)  \
0  \t\t-    Minor  Task execution failure alarm          OSS  9/17/2021 19:39   
1  \t\t-    Minor  Task execution failure alarm          OSS  9/18/2021 21:35   

   Cleared On (NT) Type Clearance Status                    Alarm Duration  
0  1/22/2022 15:16  QoS          Cleared  3043 hours 36 minutes 45 seconds  
1  1/22/2022 15:16  QoS          Cleared  3017 hours 40 minutes 50 seconds

In [10]:
print("Total Number of Rows : {}".format(df.shape[0]))
print("Total Number of Features : {}".format(df.shape[1]))

Total Number of Rows : 7427257
Total Number of Features : 7


In [11]:
# Show uniqe values of each column
for i in range(df.shape[1]):
    print("###########################     {}    ###########################".format(df.columns[i]))
    print(df.iloc[:,i].unique())
    print(" ")
    print(" ")

###########################     Severity    ###########################
['Major' 'Minor' 'Critical' 'Warning']
 
 
###########################     Name    ###########################
['IKE Negotiation Failure' 'Cell Unavailable' 'SCTP Link Congestion'
 'SCTP Link Fault' 'User Plane Path Fault' 'IP PM Activation Failure'
 'BBU CPRI Optical Module Fault' 'FAN Fault Alarm' 'IP Clock Link Failure'
 'User Plane Fault' 'X2 Interface Fault'
 'Transmission Optical Interface Error' 'BBU CPRI Interface Error'
 'Cell RX Channel Interference Noise Power Unbalanced'
 'ALD Maintenance Link Failure' 'Cabinet Humidity Unacceptable'
 'RF Unit VSWR Threshold Crossed' 'ALD Hardware Fault'
 'eNodeB S1 Control Plane Transmission Interruption'
 'External Clock Reference Problem' 'RF Unit ALD Current Out of Range'
 'No License Running in System' 'Certificate Invalid'
 'Ethernet Link Fault' 'Remote Maintenance Link Failure'
 'Time Synchronization Failure' 'RF Unit Maintenance Link Failure'
 'Data Configuratio

df.info()

In [7]:
#Check Missing Values
def check_missing(data):
    total = data.isnull().sum().sort_values(ascending=False)
    percent = (data.isnull().sum()/data.isnull().count()).sort_values(ascending=False)
    missing_data = pd.concat([total, percent], axis=1, keys=['Total', 'Percent'])
    return missing_data.head(20)

check_missing(df)

Total  Percent
Severity              0      0.0
Name                  0      0.0
Alarm Source          0      0.0
Occurred On (NT)      0      0.0
Cleared On (NT)       0      0.0
Type                  0      0.0
Alarm Duration        0      0.0

In [8]:
#check duplicate rows
def check_duplicate(data):
    duplicate = data.duplicated()
    unique_data  = pd.Series(data.shape[0] - duplicate.value_counts()[1])
    duplicate_data = pd.Series(data.shape[0] - duplicate.value_counts()[0])
    unique_data_percent = pd.Series((unique_data/data.shape[0])*100)
    duplicate_data_percent = pd.Series((duplicate_data/data.shape[0])*100)
    dub = pd.concat([unique_data, duplicate_data,unique_data_percent,duplicate_data_percent], axis=1, keys=['Unique Count', 'Duplicate Count','Unique percent','Duplicate Percent'])
    return dub

check_duplicate(df)

Unique Count  Duplicate Count  Unique percent  Duplicate Percent
0       6400938          1026319       86.181722          13.818278

# Conclusion and Recommended actions

- Convert  Occurred On (NT) & Cleared On (NT) to datetime instead of object
- there are 1026319 Duplicate rows should be dropped